### PR TITLE
Enhance vaccination allergy

### LINKED ISSUE #196: Improve vaccination allergy parsing
Allergy is always a `NONE` requirement. Make it such that the user need not specify that it is`NONE`.

### PR DESCRIPTION
**Pull request by daitenshionyan**
### Significant changes

* `allergyReq` (and related) variables renamed to `ingredients`.
* Ingredients in CLI require a list of group names instead of requirements.
* JSON format of vaccination changed to accomodate for change.

### Issues addressed

* Resolves #196

### CODE REVIEW

File: src/main/java/seedu/vms/logic/parser/vaccination/VaxTypeValueParser.java
Comment (francisyzy) [inline src/main/java/seedu/vms/logic/parser/vaccination/VaxTypeValueParser.java:1]: I think you might need to change this line

`private static final String FIELD_NAME_ALLERGY = "Allergy requirements";`

**Review by francisyzy — COMMENTED** [review]
LGTM!

Tried `vaccination add Pfizer (Dose 1) --groups DOSE 1, PFIZER, VACCINATION --lal 5 --s 56 --i allergy1, allergy2, allergy3 --h NONE::DOES 1` and it works!
Tried: 
`vaccination add Pfizer (Dose 1) --groups DOSE 1, PFIZER, VACCINATION --lal 5 --s 56 --i NONE::allergy1, allergy2, allergy3 --h NONE::DOES 1`
and it gives this error message... hmm
image

**Comment by daitenshionyan** [comment]
Its intended!! Ingredients (allergies) no longer supports the requirement syntax (ie. NONE:: ...) since its requirement can only be none. So its syntax follows the groups (--g) syntax. But I think I can improve this by giving the syntax on errors as well.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -274,30 +274,28 @@ they will be set to their default values.
...
-    --lal 5 --s 56 \
-    --a NONE::allergy1, allergy2, allergy3 \
+    --lal 5 \
+    --a allergy1, allergy2, allergy3 \
     --h NONE::DOES 1 \
 ```
 
 Copy and paste:<br>
-`vaccination add Pfizer (Dose 1) --groups DOSE 1, PFIZER, VACCINATION --lal 5 --s 56 --a NONE::allergy1, allergy2, allergy3 --h NONE::DOES 1`
+`vaccination add Pfizer (Dose 1) --groups DOSE 1, PFIZER, VACCINATION --lal 5 --a allergy1, allergy2, allergy3 --h NONE::DOES 1`
```
Comment (francisyzy) [inline docs/UserGuide.md:308]: should be `--i` ?

Comment (daitenshionyan) [reply docs/UserGuide.md]: Nice catch!!

**Review by francisyzy — APPROVED** [review]
LGTM!

**Pull request merged**